pkgs/tools/admin/pgadmin/update.sh

```diff
@@ -0,0 +1,130 @@
+pushd /tmp/pgadmin-update-script
+wget -nc $url
+tar -xzf "pgadmin4-$newest_version.tar.gz"
+cd "pgadmin4-$newest_version/web"
+
+
+######################################
+# Convert the `yarn.lock` file to v1 #
+######################################
+
+printf "${YELLOW}Will now convert the v2 lockfile. This will download the npm packages to get the metadata.\n"
+printf "Please note: This will take some time!${NC}\n"
+yarn-lock-converter -i yarn.lock -o yarn_v1.lock --cache .cache
+printf "${GREEN}Conversion done${NC}\n"
+
+###########################################################################
+# Do some post-convert corrections (namely add '"' to multipackage lines) #
+###########################################################################
+
+printf "${YELLOW}Will now do some regex substitution post-processing${NC}\n"
+sed -i -E "s|(.), |\1\", \"|g" yarn_v1.lock
+printf "${GREEN}Substituion done${NC}\n"
+
+#########################################################
+# Add the github packages we lost during the conversion #
+#########################################################
+
+printf "${YELLOW}Will now add missing github packages back to the v1 yarn.lock file${NC}\n"
+# remove header
+tail +8 yarn.lock > yarn_mod.lock
+LENGTH=$(yq '. | with_entries(select(.value.resolution | contains("github"))) | keys | length' yarn_mod.lock)
+for i in $(seq 0 $(($LENGTH-1)));
+do
+  ENTRY=$(yq ". | with_entries(select(.value.resolution | contains(\"github\"))) | keys | .[$i]" yarn_mod.lock)
+  URL=$(echo $ENTRY | cut -d "@" -f 2)
+  VERSION=$(yq ".$ENTRY.version" yarn_mod.lock)
+  LENGTH_DEP=$(yq ".$ENTRY.dependencies | keys | length" yarn_mod.lock)
+  echo "$ENTRY:" >> adendum.lock
+  echo "  version $VERSION" >> adendum.lock
+  echo "  resolved \"$URL" >> adendum.lock
+  echo "  dependencies:" >> adendum.lock
+
+  for j in $(seq 0 $(($LENGTH_DEP-1)));
+  do
+    DEPENDENCY_KEY=$(yq ".$ENTRY.dependencies | keys | .[$j]" yarn_mod.lock)
+    DEPENDENCY_VALUE=$(yq ".$ENTRY.dependencies.$DEPENDENCY_KEY" yarn_mod.lock)
+    # remove '"'
+    DEPENDENCY_KEY=${DEPENDENCY_KEY//\"}
+    echo "    \"$DEPENDENCY_KEY\" $DEPENDENCY_VALUE" >> adendum.lock
+  done
+done
+
+echo "" >> yarn_v1.lock
+cat adendum.lock >> yarn_v1.lock
+printf "${GREEN}Done${NC}\n"
+
+###########
+# cleanup #
+###########
+
+rm yarn.lock adendum.lock
+mv yarn_v1.lock yarn.lock
+
+##############################
+# Generate correct yarn hash #
+##############################
+
+printf "${YELLOW}Will now generate the hash. This will download the packages to the nix store and also take some time${NC}\n"
+YARN_HASH=$(prefetch-yarn-deps yarn.lock)
+YARN_HASH=$(nix hash to-sri --type sha256 "$YARN_HASH")
+printf "${GREEN}Done${NC}\n"
+
+##########################################
+# add the v1 `yarn.lock` file to nixpkgs #
+##########################################
+
+printf "${YELLOW}Copy files to nixpkgs${NC}\n"
+cp yarn.lock "$nixpkgs/pkgs/tools/admin/pgadmin/"
```

Comment:
```suggestion
mv yarn.lock "$nixpkgs/pkgs/tools/admin/pgadmin/"
```

Reply:
why move? It gets deleted anyway later on

Reply:
mv is generally more idempotent but not that important